### PR TITLE
Sth/expand cloud cost properties

### PR DESCRIPTION
**Pull request by Sean-Holcomb**
## What does this PR change?
This PR adds 4 new properties to the Cloud Cost type that can be aggregated or filtered on. These new fields line up with FOCUS Spec fields bringing Cloud Costs closer to the spec The fields are:

- [InvoiceEntityName](https://focus.finops.org/focus-columns/#modal-column-14231)
- [AccountName](https://focus.finops.org/focus-columns/#modal-column-14368)
- [RegionID](https://focus.finops.org/focus-columns/?prod_focus_columns%5Bquery%5D=region%20id#modal-column-14471)
- [AvailabilityZone](https://focus.finops.org/focus-columns/#modal-column-14435)

To go along with these new properties, each billing integration implementation has been updated to ingest values for them if available. When values are not available for BillingEntityName or AccountName the value from BillingEntityID and AccountID are used respectively. If there is no RegionID or AvailabilityZone value then the properties are left blank.

This Pr also contains fixes for the S3Select integration for AWS.

## Does this PR relate to any other PRs?
* 

## How will this PR impact users?
* 

## Does this PR address any GitHub or Zendesk issues?
* Closes ...

## How was this PR tested?
* This PR was tested by ingesting CloudCost data from each provider and checking that new fields were populated as they should be. Additional testing was done to ensure that old bingen files were still valid.
* Unit Tests pending

## Does this PR require changes to documentation?
* This PR requires updates to CloudCost documentation.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
*

### CODE REVIEW

File: core/pkg/filter/cloudcost/fields.go [hunk 1]
```diff
@@ -8,11 +8,15 @@ import (
...
-	FieldProviderID      CloudCostField = CloudCostField(fieldstrings.FieldProviderID)
-	FieldCategory        CloudCostField = CloudCostField(fieldstrings.FieldCategory)
-	FieldService         CloudCostField = CloudCostField(fieldstrings.FieldService)
-	FieldLabel           CloudCostField = CloudCostField(fieldstrings.FieldLabel)
+	FieldInvoiceEntityID   CloudCostField = CloudCostField(fieldstrings.FieldInvoiceEntityID)
+	FieldInvoiceEntityName CloudCostField = CloudCostField(fieldstrings.FieldInvoiceEntityName)
+	FieldAccountID         CloudCostField = CloudCostField(fieldstrings.FieldAccountID)
+	FieldAccountName       CloudCostField = CloudCostField(fieldstrings.FieldAccountName)
+	FieldRegionID          CloudCostField = CloudCostField(fieldstrings.FieldRegionID)
+	FieldAvailabilityZone                 = CloudCostField(fieldstrings.FieldAvailabilityZone)
```
Comment (avrodrigues5) [inline core/pkg/filter/cloudcost/fields.go]: Is there a particular reason why `FieldAvailabilityZone` doesnt have string alias

**Review by avrodrigues5 — APPROVED** [review]
Need an OC maintainer to approve but this looks good to me apart from maybe a missed string alias.

File: core/pkg/opencost/cloudcostprops.go [hunk 2]
```diff
@@ -152,34 +164,46 @@ func (ccl CloudCostLabels) Intersection(that CloudCostLabels) CloudCostLabels {
...
-	Labels          CloudCostLabels `json:"labels,omitempty"`
+	ProviderID        string          `json:"providerID,omitempty"`
+	Provider          string          `json:"provider,omitempty"`
+	AccountID         string          `json:"accountID,omitempty"`
+	AccountName       string          `json:"accountName,omitempty"` // @bingen:field[version=3]
+	InvoiceEntityID   string          `json:"invoiceEntityID,omitempty"`
+	InvoiceEntityName string          `json:"invoiceEntityName,omitempty"` // @bingen:field[version=3]
+	RegionID          string          `json:"regionID,omitempty"`          // @bingen:field[version=3]
+	AvailabilityZone  string          `json:"availabilityZone,omitempty"`  // @bingen:field[version=3]
+	Service           string          `json:"service,omitempty"`
```
Comment (mbolt35) [inline core/pkg/opencost/cloudcostprops.go:175]: I'm glad you're putting this into practice. I think for years, I've been preaching "add new fields to the end" but realized recently that this wasn't a requirement after the backwards compatibility updates. 

Although, once this ordering _is_ established, you can't rearrange afterwards. That's the only remaining nuance.

**Review by mbolt35 — APPROVED** [review]
Looks great to me!

**Pull request merged**